modules/data-entry/src/main/frontend/src/vocabQuery/query.jsx

```diff
@@ -433,81 +232,73 @@ class VocabularyQuery extends React.Component {
         } else {
           suggestions.push(
             <MenuItem
-              className={this.props.classes.dropdownItem}
+              className={classes.dropdownItem}
               key={NO_RESULTS_TEXT}
-              onClick={this.props.onClick}
+              onClick={props.onClick}
               disabled={true}
             >
               {NO_RESULTS_TEXT}
             </MenuItem>
           )
         }
 
-        this.setState({
-          suggestions: suggestions,
-          suggestionsVisible: true,
-          suggestionsLoading: false,
-        });
+        setSuggestions(suggestions);
+        setSuggestionsVisible(true);
+        setSuggestionsLoading(false);
     } else {
-      this.logError("Failed to search vocabulary term");
+      logError("Failed to search vocabulary term");
     }
   }
 
   // Event handler for clicking away from the autocomplete while it is open
-  closeAutocomplete = event => {
-    if ((this.menuPopperRef && this.menuPopperRef.contains(event.target))
-      || (this.infoRef && this.infoRef.contains(event.target))
-      || this.state.browserOpened) {
+  let closeAutocomplete = event => {
+    if ((menuPopperRef && menuPopperRef.current.contains(event.target))
+      || (infoRef && infoRef.current && infoRef.current.contains(event.target))
```

Comment:
Refs are always defined, but the current value may not be, so a better check would be:

```suggestion
    if (menuPopperRef?.current?.contains(event.target)
      || infoRef?.current?.contains(event.target)
```